tests/shouldwork/LoadModules/BlockRam/DualBlockRamDefinitions.hs

```diff
@@ -0,0 +1,158 @@
+{-# OPTIONS_GHC -Wno-orphans -fconstraint-solver-iterations=0#-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+module DualBlockRamDefinitions where
+
+import qualified Prelude as P
+
+import Clash.Explicit.Prelude
+import Clash.Explicit.BlockRam
+import DualBlockRamTypes
+
+-- Creating domains A, B and C
+createDomain vSystem{vName="A", vPeriod=50000} -- fast
+createDomain vSystem{vName="B", vPeriod=100000} -- slow
+createDomain vSystem{vName="C", vPeriod=140000}--hzToPeriod 7e6} -- slower
+
+-- Clocks for domains A, B and C
+clk20 = clockGen @A
+clk10 = clockGen @B
+clk7 = clockGen @C
+
+-- Resets for domains A, B and C.
+rst20 :: Reset A
+rst20 = resetGen @A
+rst10 :: Reset B
+rst10 = resetGen @B
+rst7 :: Reset C
+rst7 = resetGen @C
+
+topEntity ::
+  ( KnownDomain domA
+  , KnownDomain domB
+  ) =>
+  -- Write Modes
+  WriteMode ->
+  WriteMode ->
+
+  -- Clocks
+  Clock  domA ->
+  Clock  domB ->
+
+  --Operations
+  Signal domA (RamOp 30 ThisOrThat) ->
+  Signal domB (RamOp 30 ThisOrThat) ->
+
+  --Output
+  ( Signal domA ThisOrThat
+  , Signal domB ThisOrThat )
+
+topEntity = trueDualPortBlockRam
+
```

Comment:
Please don't call this thing `topEntity`. That's a name we ascribe special meaning to. One of them not being polymorphic as this function is.

Reply:
Also, an unneeded blank line.